runtime/onert/core/include/ir/OperandInfo.h

```diff
@@ -117,6 +117,7 @@ class OperandInfo
 
   MemAllocType memAllocType() const { return _alloc_type; }
   void setAsConstant() { _const = true; }
+  void setAsNonConst() { _const = false; }
```

Comment:
If a constant operand is used as both output of graph and input of a node, the operand is not constant?

Reply:
Good point, I think I can work on it after getting this merged.

I'm thinking changing the node's input to the newly created constant tensor in `ConstantOutputPass`.

```
( )   : non-input/output tensor
(( )) : output tensor
[ ]   : operation

Original Model
(( Const Output )) -> [ Some Operation ] -> ( Some Tensor ) -> ...

After performing ConstantOutputPass (AS-IS)
( Const ) -> (( Output )) -> [ Some Operation ] -> ( Some Tensor ) -> ...

After performing ConstantOutputPass (TO-BE)
( Const ) -> (( Output )) 
         \-> [ Some Operation ] -> ( Some Tensor ) -> ...
```

Reply:
> Good point, I think I can work on it after getting this merged.

What if the constant operand is used in multiple backends? If so, I think `setAsNonConst()` must be called after being copied to each backend by calling `ConstantInsertionPass`. But it seems to be called before `ConstantInsertionPass` in this PR.

Reply:
Wouldn't my solution resolve that case?

```
After performing ConstantOutputPass (TO-BE)
( Const ) -> (( Output )) 
         \-> [ Some Operation ] -> ( Some Tensor ) -> ...
```

Then other operations use a constant.

Reply:
I'm not sure if this pr wouldn't cover that case. It wouldn't probably cover that case.

Can your solution cover this case?
```
( Const ) -> [ Permute ] -> (( Output )) 
         \-> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
         \-> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...
```
I think your solution should cover this case too.

Reply:
Yes, as this pass is run before lowering, so those cases are handled in the later pass `PermutationInsertionPass`.

Reply:
```
== After PermutationInsertionPass ==
( Const ) -> [ Permute ] -> (( Output ))
         \-> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
         \-> [ Permute ] -> ( Permuted Const ) -> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...
```

Does `\-> [ Permute ] -> ( Permuted Const ) -> [ Some Operation on acl_cl ]` mean that a permute node is inserted for constant operand? Is it going to run every executions? Why don't you resolve it by `ConstantInitializer`?

Reply:
```
After performing ConstantOutputPass (AS-IS)
( Const ) -> (( Output )) -> [ Some Operation ] -> ( Some Tensor ) -> ...
```

"AS-IS" means this PR? I don't find the test to verify it in this PR. So I'm not sure if this PR resolves it. Could you add a test for this case?

Reply:
> Why don't you resolve it by ConstantInitializer?

Oh I probably knew something wrong. 😢  Anyway the thing is, as far as I know, our runtime can handle this graph. What `ConstantOutputPass` does is:

1. Create a permute and const input (This PR, as-is)
2. Change operation inputs to the newly-created-const (to-be-done)

```
( Const ) -> [ Permute ] -> (( Output ))
         \-> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
         \-> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...
```

And after that, what I described about the runtime was probably wrong. It seems like `ConstantInsertionPass` generates other constants(rather than creating Permute) and then `ConstantIntializer` fills all of them, and I guess it is what you meant. It it may look like this:

```
== After PermutationInsertionPass ==
( Const ) -> [ Permute ] -> (( Output ))
         \-> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
( Const ) -> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...
```

> "AS-IS" means this PR? 

Yes.

>  I don't find the test to verify it in this PR. So I'm not sure if this PR resolves it. Could you add a test for this case?

Sure, but can I do that later? I also need to prepare some negative tests. And this is originally for #4231 .